src/assay.cc

```diff
@@ -278,7 +278,32 @@ int Assay::processURI(const char *uri, const char *protocol,
     m_collections.store("REQUEST_METHOD", protocol);
     m_collections.store("REQUEST_PROTOCOL",
         "HTTP/" + std::string(http_version));
-    m_collections.store("REQUEST_URI", uri);
+
+std::string parsedURI = uri;
+    // The more popular case is without domain
+    if (m_uri_decoded.at(1) != '/'){
```

Comment:
It will be very good to have a "sanity" validation like:

```
!m_uri_decoded.empty()
```

We may not have an element on the position "1". The empty string will lead c++ to throw an instance of 'std::out_of_range'